File: frontend/src/lib/changeStatus.ts
export type ChangeStatus =
  | "eingefuehrt"
  | "geaendert"
  | "abgeschafft"
  | "unveraendert";

export function normalizeChangeStatus(value: unknown): ChangeStatus | null {
  if (typeof value !== "string") {
    return null;
  }
  const raw = value.trim().toLowerCase();
  if (
    raw === "unveraendert" ||
    raw === "unverändert" ||
    raw === "gleich" ||
    raw === "gleichbleibend" ||
    raw === "unchanged" ||
    raw === "same" ||
    raw === "no change" ||
    raw === "no_change"
  ) {
    return "unveraendert";
  }
  if (raw === "eingefuehrt" || raw === "eingeführt" || raw === "introduced" || raw === "new") {
    return "eingefuehrt";
  }
  if (
    raw === "abgeschafft" ||
    raw === "entfallen" ||
    raw === "entfaellt" ||
    raw === "obsolete" ||
    raw === "deleted" ||
    raw === "removed"
  ) {
    return "abgeschafft";
  }
  if (raw === "geaendert" || raw === "geändert" || raw === "changed" || raw === "updated") {
    return "geaendert";
  }
  return null;
}

export function getChangeStatusLabel(status: ChangeStatus): string {
  if (status === "eingefuehrt") {
    return "Neu";
  }
  if (status === "abgeschafft") {
    return "Abgeschafft";
  }
  if (status === "unveraendert") {
    return "Unverändert";
  }
  return "Geändert";
}

File: backend/core/change_status.py
from __future__ import annotations

from collections.abc import Mapping


CHANGE_STATUS_KEYS = ("aenderungsstatus", "change_status", "status_change", "status")


def normalize_change_status(value: object) -> str:
    raw = str(value or "").strip().lower()
    if raw in {
        "unveraendert",
        "unverändert",
        "gleich",
        "gleichbleibend",
        "unveraendert geblieben",
        "unchanged",
        "same",
        "no_change",
        "no change",
    }:
        return "unveraendert"
    if raw in {"eingefuehrt", "eingeführt", "introduced", "new"}:
        return "eingefuehrt"
    if raw in {"abgeschafft", "entfallen", "entfaellt", "obsolete", "deleted", "removed"}:
        return "abgeschafft"
    if raw in {"geaendert", "geändert", "changed", "updated"}:
        return "geaendert"
    return "geaendert"


def extract_change_status(payload: Mapping[str, object]) -> str:
    for key in CHANGE_STATUS_KEYS:
        value = payload.get(key)
        if value is None:
            continue
        return normalize_change_status(value)
    return "geaendert"
